src/class_sqlThread.py

```diff
@@ -17,128 +17,100 @@
     import state
     from addresses import encodeAddress
     from bmconfigparser import config, config_ready
-    from debug import logger
     from tr import _translate
 except ImportError:
     from . import helper_sql, helper_startup, paths, queues, state
     from .addresses import encodeAddress
     from .bmconfigparser import config, config_ready
-    from .debug import logger
     from .tr import _translate
 
+logger = logging.getLogger('default')
 
-class sqlThread(threading.Thread):
-    """A thread for all SQL operations"""
 
-    def __init__(self):
-        threading.Thread.__init__(self, name="SQL")
+class BitmessageDB(object):
+    """ Upgrade Db with respect to versions """
 
-    def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
-        """Process SQL queries from `.helper_sql.sqlSubmitQueue`"""
-        helper_sql.sql_available = True
-        config_ready.wait()
-        self.conn = sqlite3.connect(state.appdata + 'messages.dat')
+    def __init__(self):
+        self._current_level = None
+        self.max_level = 11
+        self.conn = None
+        self.cur = None
+        self._connection_build()
+        self.root_path = os.path.dirname(os.path.dirname(__file__))
+
+    def _connection_build(self):
+        self._connection_build_internal('messages.dat', False)
+
+    def _connection_build_internal(self, file_name="messages.dat", memory=False):
+        """
+            Stablish SQL connection
+        """
+        if memory:
+            self.conn = sqlite3.connect(':memory:')
+        else:
+            self.conn = sqlite3.connect(os.path.join(state.appdata + file_name))
         self.conn.text_factory = str
         self.cur = self.conn.cursor()
-
         self.cur.execute('PRAGMA secure_delete = true')
 
-        # call create_function for encode address
-        self.create_function()
-
+    def __get_current_settings_version(self):
+        """
+            Get current setting Version
+        """
+        query = "SELECT value FROM settings WHERE key='version'"
+        parameters = ()
+        self.cur.execute(query, parameters)
+        return int(self.cur.fetchall()[0][0])
+
+    def _upgrade_one_level_sql_statement(self, file_name):
+        """
+            Upgrade database versions with applying sql scripts
+        """
+        self.initialise_sql("init_version_{}".format(file_name))
+
+    def initialise_sql(self, file_name):
```

Comment:
the PR should use consistent language dialect. In some places in this PR verbs end with `-se` (UK) and other places with `-ze` (US). I don't think the project has a standard for this but I think it should be `-ze`.